src/poprox_recommender/components/filters/image_selector.py

```diff
@@ -0,0 +1,66 @@
+import logging
+from typing import Optional
+
+import torch as th
+
+from poprox_concepts.domain import Image, InterestProfile, RecommendationList
+from poprox_recommender.components.embedders.caption_embedder import CaptionEmbedder
+from poprox_recommender.pytorch.decorators import torch_inference
+
+logger = logging.getLogger(__name__)
+
+
+class ImageSelector:
+    caption_embedder: CaptionEmbedder
+
+    def __init__(self, caption_embedder: CaptionEmbedder):
+        self.caption_embedder = caption_embedder
+
+    @torch_inference
+    def __call__(self, recommendations: RecommendationList, interest_profile: InterestProfile) -> RecommendationList:
+        if not recommendations.articles:
+            # logger.debug("No articles in recommendation list")
+            return recommendations
+
+        if interest_profile.embedding is None:
+            # logger.warning("No user embedding available; skipping image selection")
+            return recommendations
+
+        user_embedding = interest_profile.embedding.squeeze(0)
+        # logger.debug(f"User embedding shape: {user_embedding.shape}")
+
+        for article in recommendations.articles:
+            selected_image = self._select_article_image(article, user_embedding)
+            if selected_image:
+                article.preview_image_id = selected_image.image_id
+                # logger.debug(f"Selected image {selected_image.image_id} for article {article.article_id}")
+
+        return recommendations
+
+    def _select_article_image(self, article, user_embedding) -> Optional[Image]:
```

Comment:
Adding type annotations would make this a bit easier to read
```suggestion
    def _select_article_image(self, article: Article, user_embedding: th.Tensor) -> Image | None:
```
The `Optional[Image]` type is correct but you can do it without importing from `typing` with the `A | B` syntax